### PR TITLE
[core] move transformers to setup_utils, bump dependency version

### PR DESCRIPTION
**Pull request by canerturkmen**
*Issue #, if available:*

*Description of changes:*

Chronos implementation (cf #3978) requires `transformers>=4.36`. This PR moves transformers and accelerate version ranges to `_setup_utils.py` and bumps the version. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by canerturkmen** [comment]
In Chronos, we would like to depend on `optimum-intel` for fast CPU inference, and this requires `transformers>=4.36`. We can ship without `optimum-intel` support too, but this results in 2-3x slower inference on some models. 

Currently, bumping the `transformers` version appears to be breaking a specific unit test. If this is easy to mitigate for MM, then we can include that fix here or in another PR to enable Chronos.

Looking forward to your comments.

**Comment by tonyhoo** [comment]
@prateekdesai04  we should test this change with torch update

File: timeseries/setup.py [hunk 1]
```diff
@@ -55,7 +57,9 @@
     ],
 }
 
-extras_require["all"] = []
+extras_require["all"] = [
+    "optimum[openvino,onnxruntime,nncf]>=1.17,<1.18",  # for faster CPU inference in pretrained models
```
Comment (tonyhoo) [inline timeseries/setup.py:61]: Do we need [neuron suppport](https://huggingface.co/docs/optimum-neuron/index) to speed up inference on Inf2 architecture?

**Comment by prateekdesai04** [comment]
> @prateekdesai04 we should test this change with torch update

@tonyhoo Just pushed the latest Torch + CUDA image, running this PR https://github.com/autogluon/autogluon/pull/3982
Once successful we can merge it and then run Caner's PR with the latest changes

File: timeseries/setup.py (hunk 1 above)
Comment (canerturkmen) [reply timeseries/setup.py:61]: Not at the moment, though this will come in v1.2.

**Comment by canerturkmen** [comment]
Converted to draft, pending the PyTorch upgrade.

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |

**Comment by canerturkmen** [comment]
@zhiqiangdon for your review of test fails

File: core/src/autogluon/core/_setup_utils.py [hunk 2]
```diff
@@ -30,6 +30,8 @@
     "lightning": ">=2.1,<2.2",  # Major version cap
     "pytorch_lightning": ">=2.1,<2.2",  # Major version cap, capping `lightning` does not cap `pytorch_lightning`!
     "async_timeout": ">=4.0,<5",  # Major version cap
+    "transformers[sentencepiece]": ">=4.36.0,<4.39.0",
```
Comment (zhiqiangdon) [inline core/src/autogluon/core/_setup_utils.py:33]: Why do we use 4.36 instead of 4.38 as the lower cap?

Comment (canerturkmen) [reply core/src/autogluon/core/_setup_utils.py:33]: This is the lowest required by optimum for now. Wanted to leave it wide for more flexibility.

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |

**Review by tonyhoo — APPROVED** [review]

**Review by zhiqiangdon — APPROVED** [review]
LGTM

**Pull request merged**